pandas/core/indexes/range.py

```diff
@@ -344,12 +345,14 @@ def __contains__(self, key: Union[int, np.integer]) -> bool:
 
     @Appender(_index_shared_docs["get_loc"])
     def get_loc(self, key, method=None, tolerance=None):
-        if is_integer(key) and method is None and tolerance is None:
-            new_key = int(key)
-            try:
-                return self._range.index(new_key)
-            except ValueError:
-                raise KeyError(key)
+        if method is None and tolerance is None:
+            if is_integer(key) or (is_float(key) and key.is_integer()):
```

Comment:
Does the extra condition here imply any behavioral changes?

Reply:
No, intlike floats go through here now instead of going through the _maybe_cast_indexer path in Index.get_loc.